enumerable.rb

```diff
@@ -0,0 +1,89 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    size.times { |n| yield (self[n]) }
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    size.times { |n| yield self[n], n }
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result_array = []
+    my_each { |n| result_array << n if yield(n) }
+    result_array
+  end
+
+  def my_all
+    return true unless block_given?
+
+    my_each { |n| return false unless yield(n) }
+    true
+  end
+
+  def my_any
+    return true unless block_given?
+
+    my_each { |n| return true if yield(n) }
+    false
+  end
+
+  def my_none
+    return true unless block_given?
```

Comment:
### **my_none**

- [ ] It should return true if the block never returns true for all elements.

- [ ] It should not mutate the original array

- [ ] when no block or argument is given it should return true only if none of the collection members is true

- [ ] when a class is passed as an argument it should return true if none of the collection is a member of such class

- [ ] when a Regex is passed as an argument it should return true only if none of the collection matches the Regex

Reply:
fixed the logic, all ruby doc test passed 
![passed2](https://user-images.githubusercontent.com/51087315/83947787-e229d180-a821-11ea-9127-e45c8e242232.png)